### ISSUE TITLE
Error in assignment?

### ISSUE THREAD

**Issue by benjeffery** [tskit-dev/tsdate#208]
Whilst updating pre-commit (#207) it detected this line: `tsdate/prior.py:658`

Here's the surrounding snippet, it looks like a mistake:

```python
# Add unary nodes below the altered ones, as their result is calculated
            # from the coalescent node above
            unary_descendants = set()
            for node in changed_nodes:
                children = prev_tree.children(node)
                if children is not None:
                    if len(children) == 1:
                        # Keep descending
                        node == children[0]   <-------- NOTE ==
                        while True:
                            children = prev_tree.children(node)
                            if len(children) != 1:
                                break
                            unary_descendants.add(node)
                            node = children[0]
                    else:
                        # Descend all branches, looking for unary nodes
                        for node in prev_tree.children(node):
                            while True:
                                children = prev_tree.children(node)
                                if len(children) != 1:
                                    break
                                unary_descendants.add(node)
                                node = children[0]
```

**Comment by awohns** [comment]
Thanks @benjeffery, @hyanwong would know best, but that looks ok to me, since it checks that children is of length 1 before running `children[0]`?

**Comment by benjeffery** [comment]
It's not `children[0]` that is the issue, it is the `==` making the line a noop!

**Comment by awohns** [comment]
Oh! Yes good catch

**Comment by benjeffery** [comment]
Interestingly the tests still pass if I replace it with `=` so I wonder what the performance effect is.

**Comment by benjeffery** [comment]
Looking at the code, I don't think it will change much as in the inner while loop there is a descent.

**Comment by benjeffery** [comment]
@hyanwong @jeromekelleher Any thoughts as to whether this line should be deleted or fixed to be an assignment rather than a comparison?

**Comment by hyanwong** [comment]
Just looking at it now.

**Comment by awohns** [comment]
Any thoughts @hyanwong ?

**Comment by hyanwong** [comment]
Sorry, deep in the ARG paper.

I would like to fix the node metadata struct before a release, as this would be a breaking change. See also https://github.com/tskit-dev/tsinfer/pull/642

**Comment by benjeffery** [comment]
@hyanwong Sorry to bump, but would be good to close this out. I think the line can just be deleted?

**Comment by benjeffery** [comment]
I've merged #207 as it isn't a behavior change. Close this issue if you're happy that nothing else needs to change @hyanwong.

**Comment by hyanwong** [comment]
I think we can simply remove the line. If we change to an assignment then we wouldn't allocate the current node (if unary) to `unary_descendants`.

This is all slightly moot, as we don't usually allow dating of tree sequences with unary nodes anyway!
